geos: bump to include seg fault fix

otan opened the pull request:
Resolves #53254.

Release justification: bug fix to new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53907)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53907#-:-MGK5QvBAmhER4wKIlOS:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[c-deps/geos, line 1 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53907#-MGK5GP7BefCrgodWzqL:-MGK5GP8EM8Euz6F0Dx9:b-ezqt2t) ([raw file](https://github.com/cockroachdb/cockroach/blob/7554c94f4fbbf276f57fb34a97919faf9bf0f32b/c-deps/geos#L1)):*
> ```
> Subproject commit 407f899d250614dd38bdb6d6a586d8fc4066ba17
> ```

the commit (after the tweak to avoid casting again) is 4a...
https://github.com/cockroachdb/geos/commit/4a98497529eb6f5a07d334583962bdffd65dd380


<!-- Sent from Reviewable.io -->

otan committed:
geos: bump to include seg fault fix

Resolves #53254.

Release justification: bug fix to new functionality

Release note: None

otan commented:
bors r=sumeerbhola
